patcher/src/ts/sidebar/ServerSelect.tsx

```diff
@@ -49,66 +54,171 @@ class ActiveServerView extends React.Component<ActiveServerViewProps, ActiveServ
 };
 
 export interface ServerListViewState {};
 class ServerListView extends React.Component<ServerListViewProps, ServerListViewState> {
   render() {
-    let totalPlayers = (this.props.item.arthurians|0) + (this.props.item.tuathaDeDanann|0) + (this.props.item.vikings|0);
-    let status = this.props.item.playerMaximum > 0 ? 'online' : 'offline';
-    let accessLevel = AccessType[this.props.item.accessLevel];
+    let totalPlayers = (this.props.item.serverInfo.arthurians|0) + (this.props.item.serverInfo.tuathaDeDanann|0) + (this.props.item.serverInfo.vikings|0);
+    let status = this.props.item.serverInfo.playerMaximum > 0 ? 'online' : 'offline';
+    let accessLevel = AccessType[this.props.item.serverInfo.accessLevel];
+
     return (
       <div className='server-select quickselect-list'>
         <div>
           <div className='server-status'><div className={'indicator ' + status} data-position='right'
             data-delay='150' data-tooltip={status} /></div>
           <div className='server-details'>
-            <h5 className='server'>{this.props.item.name} ({accessLevel})</h5>
-            <h6 className='server-players'>Players Online: {totalPlayers}/{this.props.item.playerMaximum}</h6>
+            <h5 className='server'>{this.props.item.displayName} ({accessLevel})</h5>
+            <h6 className='server-players'>Players Online: {totalPlayers}/{this.props.item.serverInfo.playerMaximum}</h6>
           </div>
         </div>
       </div>
     );
   }
 }
 
+interface IServerOption {
+  displayName: string;
+  serverInfo: Server;
+  channelInfo: Channel;
+}
+
+function mapToProps(state: any): any {
+  return {
+    channelsState: state.channels,
+    serversState: state.servers
+  }
+}
+
 export interface ServerSelectProps {
-  servers: Array<Server>;
-  selectedServer: Server;
-  onSelectedServerChanged: (server: Server) => void;
+  dispatch?: (action: any) => void;
+  channelsState?: ChannelState;  
+  serversState?: ServersState;
 };
 
 export interface ServerSelectState {
 };
 
 class ServerSelect extends React.Component<ServerSelectProps, ServerSelectState> {
   public name: string = 'cse-patcher-server-select';
+  private mergedServerList: { [key:string] : IServerOption } = {};
+  
+  private listAsArray:Array<IServerOption> = null;
 
   constructor(props: ServerSelectProps) {
     super(props);
   }
 
-  onSelectedServerChanged = (server: any) => {
-    this.props.onSelectedServerChanged(server);
+  onSelectedServerChanged = (server: IServerOption) => {
+    const { dispatch } = this.props;
+
+    events.fire('play-sound', 'select');
+
+    //is there a required order here?
+    dispatch(changeChannel(server.channelInfo));
+    dispatch(changeServer(server.serverInfo));
+    dispatch(fetchCharacters());
+    dispatch(selectCharacter(null));
   }
 
-  generateActiveView = (server: any) => {
-    return <ActiveServerView item={server} />
+  generateActiveView = (server: IServerOption) => {
+    if(server.serverInfo)
```

Comment:
https://github.com/airbnb/javascript#blocks--braces

    if (expression) statement;

or

    if (expression) {
      statement;
    }

are the acceptable forms.